### PR TITLE
Wifi updates

### PR DESCRIPTION
**Pull request by mikedarzen**
If WiFi is not connected at startup (or the known network is not found) then the beacon enters into WiFi setup mode.   In this mode, the "BeaconOfGondor" wifi network is created and the beacon acts as an AP.  Connect to the network, and then look at your IP address.   Visit the gateway address in a web browser and enter your wifi network information and socketio server address.  Submit and this saves to eeprom for use on next startup.   Restart the beacon and you should now be connected to the network previously specified.

### CODE REVIEW

**Review by NicJudsonZenPlanner — APPROVED** [review]

File: ESP8266 Beacon Client/BeaconClient-V2.ino [hunk 1]
```diff
@@ -79,30 +82,37 @@ void connectToNetwork() {
...
+      delay(500);
+    }
+    initializeSocketIo();
+  } else {
+    setupWifi();
   }
 }
 
+void initializeSocketIo() {
+  MDNS.begin("esp8266");
```
Comment (petrig) [inline ESP8266 Beacon Client/BeaconClient-V2.ino:106]: What is this constant?

File: ESP8266 Beacon Client/BeaconClient-V2.ino [hunk 2]
```diff
@@ -111,13 +121,16 @@ void setupWifi() {
...
   server.on("/", handleRoot);
+  server.on("/network", handleNewNetwork);
   server.begin();
 }
 
 void loop() {
   if (settingUpWifi) {
     server.handleClient();    
   } else {
+    if (!urlConnected && connectionTimer < millis())
```
Comment (petrig) [inline ESP8266 Beacon Client/BeaconClient-V2.ino:132]: Prefer braces even if only one line.

File: ESP8266 Beacon Client/handleHttp.ino [hunk 3]
```diff
@@ -1,4 +1,30 @@
...
+  EEPROM.get(0, wifiAuthPair); 
+  char html[512];
+  char url[256];
+  EEPROM.get(sizeof(wifiAuthPair), url);
+  sprintf(html,"<h1>Current SSID: %s</h1><br /><h1>Current URL: %s</h1><br /><form action=\"/network\" method=\"POST\">SSID: <input type=\"text\" name=\"ssid\" value=\"\"></br>Password: <input type=\"text\" name=\"password\" value=\"\"></br>URL: <input type=\"text\" name=\"url\" value=\"\"></br><input type=\"submit\" value=\"Submit\"></form>",wifiAuthPair.ssid, url);
+  server.send(200, "text/html", html);
+}
+
+void handleNewNetwork() {
+  if (server.hasArg("ssid")== false){
```
Comment (petrig) [inline ESP8266 Beacon Client/handleHttp.ino:13]:
```suggestion
  if (server.hasArg("ssid") == false) {
```

File: ESP8266 Beacon Client/BeaconClient-V2.ino [hunk 4]
```diff
@@ -79,30 +82,37 @@ void connectToNetwork() {
...
-  if (!networkFound)
-    softReset();
-
-  WiFi.begin(ssid[n], password[n]);
-  while (WiFi.status() != WL_CONNECTED) {
-    delay(500);
+  if (networkFound) {
+    WiFi.begin(wifiAuthPair.ssid, wifiAuthPair.password);
+    while (WiFi.status() != WL_CONNECTED) {
+      delay(500);
```
[the remark this answers is not in the document]
Comment (mikedarzen) [reply ESP8266 Beacon Client/BeaconClient-V2.ino:97]: Conceivably,  Maybe if the SSID is correct but the password is wrong.  Maybe I'll convert this to the timeout and reset in setup WiFi mode if it times out.

File: ESP8266 Beacon Client/BeaconClient-V2.ino (hunk 1 above)
Comment (mikedarzen) [reply ESP8266 Beacon Client/BeaconClient-V2.ino:106]: esp8266 is the WiFi hardware that I'm using.  I believe this statement enables dns resolution for it.

**Pull request merged**